charmcraft/services/charmlibs.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+"""Service class for interacting with charm libraries."""
+from __future__ import annotations
+
+import pathlib
+
+import craft_application
+
+from charmcraft import models, utils
+from charmcraft.store.models import Library
+
+
+class CharmLibsService(craft_application.ProjectService):
+    """Business logic for creating packages."""
+
+    _project: models.CharmcraftProject  # type: ignore[assignment]
+
+    def __init__(
+        self,
+        app: craft_application.AppMetadata,
+        services: craft_application.ServiceFactory,
+        *,
+        project: models.CharmcraftProject,
+        project_dir: pathlib.Path,
+    ) -> None:
+        super().__init__(app, services, project=project)
+        self._project_dir = project_dir
+
+    def is_lib_downloaded(
```

Comment:
is the `_lib` superfluous in these three methods?

At the end of the day you will be doing `self._services.charm_libs.[get_version,is_downloaded,write]`

Reply:
Agreed! One other change - I replaced `get_lib_version` with `get_local_version` to make it more obvious that this is checking only locally and not making any requests to the store.